user-guide/Troubleshooting/Procedures/Troubleshooting_Cassandra.md

```diff
@@ -0,0 +1,157 @@
+---
+uid: Troubleshooting_Cassandra
+---
+
+# Troubleshooting – Cassandra
+
+## Context
+
+DataMiner uses Cassandra in various ways. Each way has its specific behavior and place within a DataMiner System.
+
+Two types of Cassandra storage are used, i.e. two distinct types of Cassandra databases:
+
+- Dedicated clustered storage:, called *Cassandra Cluster* below.
+- Storage per DMA (with or without indexing), called *Cassandra Single* below.
+
+> [!TIP]
+> For more information on available storage options, see [About databases](xref:Databases_about).
+
+When you start troubleshooting, you first need to determine whether the setup uses *Cassandra Cluster* or *Cassandra Single*. There are some key differences between the two that are important for your investigation.
+
+To do so, in DataMiner Cube, go to *System Center* > *Database* > *Type*:
+
+- If the type is *database per cluster*, *Cassandra Cluster* (i.e. *dedicated clustered storage*) is used.
+- If the type is *Cassandra*, *Cassandra Single* (i.e. *Storage Per DMA*) is used.
+
+## Architecture
+
+- *Cassandra Single*:
+
+  - In case of a Failover system, a healthy *Cassandra Single* system has two clustered nodes with replication factor "2" and replication strategy "simple strategy".
+
+  - If there are no Failover Agents, each Agent is a single Cassandra node that is not clustered with the other Cassandra nodes linked to DataMiner.
+
+- *Cassandra Cluster*: The number of nodes depends on the configuration.
+
+## Issue symptoms
+
+- DataMiner fails to start up. Errors can be found in the *SLDBConnection.txt* log file.
+
+- Elements fail to start and are in status "Error".
+
+- In the Alarm Console, the following alarms are shown:
+
+  - Error alarm with value "General database failure, failed to initialize..."
+  - Error alarm mentioning "DataMiner goes into offload mode"
+
+    > [!NOTE]
+    > This error may also be caused by other databases (e.g. Elasticsearch).
```

Comment:
Maybe we should change this to OpenSearch, since Elasticsearch is no longer recommended?

Reply:
Can we both since we still have setups using Elasticsearch. And issues are more likely to occur with something old / no longer recommended. For example:
e.g. OpenSearch, Elasticsearch. Note that the latter is no longer recommended.